.github/workflows/_test_build_fake_prerelease.yaml

```diff
@@ -1,64 +1,71 @@
-name: Test Build Windows
+name: Test Build *Nix
 
 on:
-  workflow_run:
-    workflows: [ "Test Build Nix fake prerelease" ]
-    types: [completed]
   workflow_call:
     inputs:
-      original_repo_name:
-        required: true
-        type: string
-      repo_full_name:
-        required: true
-        type: string
-      tag:
-        required: true
-        type: string
       integration:
         required: true
         type: string
     secrets:
-      pfx_certificate_base64:
-        description: "pfx base64 certificate"
+      OHAI_PFX_CERTIFICATE_BASE64:
         required: true
-      pfx_passphrase:
-        description: "pfx passphrase"
+      OHAI_PFX_PASSPHRASE:
         required: true
 
 jobs:
+  test-build-nix:
+    name: Test binary compilation for all platforms:arch
+    runs-on: ubuntu-20.04
+    env:
+      INTEGRATION: ${{ inputs.integration }}
+      REPO_FULL_NAME: ${{ github.event.repository.full_name }}
+      TAG: "v0.0.0"
+    steps:
+      - uses: actions/checkout@v4
+      - run: |
+          git tag "$TAG"
+          if [ -z "$GPG_PASSPHRASE" ]; then
+            echo NO_SIGN=true >> $GITHUB_ENV
+          fi
+      - name: Build all platforms:arch
+        run: make ci/fake-prerelease
+      - name: Upload artifacts for next job
+        uses: actions/upload-artifact@v3
+        with:
+          name: windows-packages
+          path: dist/nri-*.zip
+
   test-build-windows:
-    if: ${{ github.event.workflow_run.conclusion == 'success' }}
```

Comment:
shouldn't be this windows job optional, for integrations that don't support Win?

Reply:
Yes, it's at the workflow level. For integrations that don't support Win, the build_nix workflow will be called instead of the prerelease one. You can check how this is done in the PRs I shared in the description, LMKWYT.